src/usr/share/gpgmailer/gpgmailer.py

```diff
@@ -25,74 +25,90 @@
 
         self.logger.info('Done initializing gpgmailer module.')
 
-    # Gpgmailer's main loop. Reads the watch directory and then calls other modules
-    #   to build and send email.
+
+    # GpgMailer's main program loop. Reads the watch directory and then calls other modules
+    #   to build and send e-mail. Also sends warnings about GPG key expirations.
     def start_monitoring(self):
         try:
             while True:
                 loop_start_time = time.time()
 
-                self.recipients = self.gpgkeyverifier.get_valid_recipients(loop_start_time)
-                self.keys = self.gpgkeyverifier.get_valid_keys(loop_start_time)
+                self.valid_recipient_emails =
+                    self.gpgkeyverifier.get_valid_recipient_emails(loop_start_time)
+                self.valid_key_fingerprints =
+                    self.gpgkeyverifier.get_valid_key_fingerprints(loop_start_time)
 
-                new_expiration_warning_message = self.gpgkeyverifier.get_expiration_warning_message(loop_start_time)
+                new_expiration_warning_message =
+                    self.gpgkeyverifier.get_expiration_warning_message(loop_start_time)
                 self._update_expiration_warning_message(new_expiration_warning_message)
                 self._send_email_if_new_expiration_warning_messages()
 
                 for file_name in self._get_file_list():
-                    self.logger.info("Found queued email in file %s." % file_name)
+                    self.logger.info("Found queued e-mail in file %s." % file_name)
                     message_dict = self._read_message_file(file_name)
 
 
                     # Set default subject if the queued message does not have one.
-                    if message_dict['subject'] == None:
+                    if message_dict['subject'] is not None:
                         message_dict['subject'] = self.config['default_subject']
 
-                    encrypted_message = self._build_encrypted_message(loop_start_time, message_dict)
+                    encrypted_message = self._build_encrypted_message(message_dict, loop_start_time)
 
-                    self.mailsender.sendmail(message_string=encrypted_message, recipients=self.recipients)
+                    self.mailsender.sendmail(message_string=encrypted_message,
+                        recipients=self.valid_recipient_emails)
                     self.logger.info('Message %s sent successfully.' % file_name)
 
                     os.remove(os.path.join(self.outbox_path, file_name))
 
                 time.sleep(self.config['main_loop_delay'])
 
-        except gpgkeyverifier.NoUsableKeysException as e:
-            self.logger.critical('No keys available for encryption. Exiting.')
+        except gpgkeyverifier.NoUsableKeysException as exception:
+            self.logger.critical('No keys available for encryption. Exiting. %s: %s' % 
+                (type(exception).__name__, exception.message))
```

Comment:
The exception message is almost identical to this log message. Is it helpful to print it?